pulsar-broker/src/main/java/org/apache/pulsar/broker/admin/impl/NamespacesBase.java

```diff
@@ -799,27 +799,22 @@ protected void internalSetAutoTopicCreation(AsyncResponse asyncResponse,
                         "Number of partitions should be less than or equal to " + maxPartitions);
             }
         }
-        // Force to read the data s.t. the watch to the cache content is setup.
-        namespaceResources().setAsync(path(POLICIES, namespaceName.toString()), (policies) -> {
-            policies.autoTopicCreationOverride = autoTopicCreationOverride;
-            return policies;
-        }).thenApply(r -> {
+        try {
```

Comment:
now that we fixed setAsync there is no need to change this block.
also it is better to keep the async style

the same applies to the other `updatePolicies`. calls